libflux/src/ast/mod.rs

```diff
@@ -96,62 +96,62 @@ where
 #[derive(Debug, PartialEq, Clone, Serialize, Deserialize)]
 #[serde(untagged)]
 pub enum Expression {
-    Idt(Identifier),
-
-    Arr(Box<ArrayExpression>),
-    Fun(Box<FunctionExpression>),
-    Log(Box<LogicalExpression>),
-    Obj(Box<ObjectExpression>),
-    Mem(Box<MemberExpression>),
-    Idx(Box<IndexExpression>),
-    Bin(Box<BinaryExpression>),
-    Un(Box<UnaryExpression>),
-    Pipe(Box<PipeExpression>),
-    Call(Box<CallExpression>),
-    Cond(Box<ConditionalExpression>),
-    StringExp(Box<StringExpression>),
-    Paren(Box<ParenExpression>),
-
-    Int(IntegerLiteral),
-    Flt(FloatLiteral),
-    Str(StringLiteral),
-    Dur(DurationLiteral),
-    Uint(UnsignedIntegerLiteral),
-    Bool(BooleanLiteral),
-    Time(DateTimeLiteral),
-    Regexp(RegexpLiteral),
-    PipeLit(PipeLiteral),
-
-    Bad(Box<BadExpression>),
+    Identifier(Identifier),
+
+    Array(Box<ArrayExpr>),
+    Function(Box<FunctionExpr>),
+    Logical(Box<LogicalExpr>),
+    Object(Box<ObjectExpr>),
+    Member(Box<MemberExpr>),
+    Index(Box<IndexExpr>),
+    Binary(Box<BinaryExpr>),
+    Unary(Box<UnaryExpr>),
+    PipeExpr(Box<PipeExpr>),
+    Call(Box<CallExpr>),
+    Conditional(Box<ConditionalExpr>),
+    StringExpr(Box<StringExpr>),
+    Paren(Box<ParenExpr>),
+
+    Integer(IntegerLit),
+    Float(FloatLit),
+    StringLit(StringLit),
```

Comment:
`String(StringLit)`?

Reply:
There are StringLit and StringExpr types so those would conflict so I used the long names on those constructors. Same goes for PipeLit and PipeExpr.